packages/appkit/src/utils/HelpersUtil.ts

```diff
@@ -239,7 +239,7 @@ export const WcHelpersUtil = {
         if (regex.test(currentOrigin)) {
           return true
         }
-      } else if (pattern === currentOrigin) {
+      } else if (new URL(pattern).origin === currentOrigin) {
```

Comment:
Can this break with a non url like pattern?

Reply:
When the url has asterix (which is a pattern, it's handled in the first if clause